src/repositories/story.repository.ts

```diff
@@ -0,0 +1,85 @@
+import { Repository } from '../core/repository';
+import { SUPPORTED_CAPABILITIES } from '../core/constants';
+import {
+  StoryRepositoryResponseRootObject,
+  StoryRepositoryResponseReels,
+  StoryRepositoryResponseItem,
+} from '../responses/story.repository.response';
+import { Reel } from '../types/story.reel';
+
+export class StoryRepository extends Repository {
+  async reelsMediaFeed(
+    user_ids: Array<number | string>,
+    source: string = 'feed_timeline',
+  ): Promise<StoryRepositoryResponseReels> {
+    const { body } = await this.client.request.send<StoryRepositoryResponseRootObject>({
+      url: `/api/v1/feed/reels_media/`,
+      method: 'POST',
+      form: this.client.request.signPost({
+        user_ids,
+        source,
+        _uuid: this.client.state.uuid,
+        _uid: await this.client.state.extractCookieAccountId(),
+        _csrftoken: this.client.state.CSRFToken,
+        device_id: this.client.state.deviceId,
+        supported_capabilities_new: JSON.stringify(SUPPORTED_CAPABILITIES),
+      }),
+    });
+    return body.reels;
+  }
+
+  async markSeen(reels: Reel, module: string = 'feed_timeline'): Promise<any> {
+    const { body } = await this.client.request.send<any>({
+      url: `/api/v2/media/seen/`,
+      method: 'POST',
+      form: this.client.request.signPost({
+        reels,
+        container_module: module,
+        reel_media_skipped: [],
+        live_vods: [],
+        live_vods_skipped: [],
+        nuxes: [],
+        nuxes_skipped: [],
+        reel: 1,
+        live_vod: 0,
+        _uuid: this.client.state.uuid,
+        _uid: await this.client.state.extractCookieAccountId(),
+        _csrftoken: this.client.state.CSRFToken,
+        device_id: this.client.state.deviceId,
+      }),
+    });
+    return body;
+  }
+
+  formatReels(items: StoryRepositoryResponseItem[], sourceId: string = null) {
```

Comment:
What does this function do? Is that public or private? What type does it return?

Reply:
It returns object of formatted reels from user story feed for `markSeen` request. 
It is helper method. I don't know where to write helpers, correct me.

Get user stories, then format it, then mark it as seen.
```
return await ig.story.reelsMediaFeed([userId])
    .then(result => {
      const reels = ig.story.formatReels(result[userId].items);
      return ig.story.markSeen(reels);
    });
```

Reply:
Got it.
To be honest, i don't think repository class is the right place for such type of functions.
Right now i'm implementing active-record classes, i'll try to introduce it tomorrow.
It will be look approximately like this
```
class StoryReel {
  get markSeenObject() {
    // Your logic here
  }
}
```

Reply:
ok, I'll wait for the  active-record then do code refactoring and commit all changes.